### ISSUE TITLE
Add specification for the `/health` endpoint

### ISSUE THREAD

**Issue by binarylogic** [vectordotdev/vector#10556]
#4250 and, subsequently, #9460 both suffer from a vague definition of the `/health` endpoint. To guide that work, we should take time to define exactly what the `/health` endpoint is used for and how it should respond under various failure modes.

Therefore, I propose that we add a new `docs/specs/api.md` with `[GET] /health` as a section in the specification. It should:

1. Describe exactly what the endpoint is used for
2. Document the response status codes and what they mean.
3. Document the response payloads.
   1. I'd like to see individual component health as part of the payload (see #10555)

#10555 should make the work here much more clear.

**Comment by spencergilbert** [comment]
This ignores the existence of the "health" GraphQL query which is separate. How does that fit into this issue?

**Comment by binarylogic** [comment]
Yes, those should be unified and response similarly. That is in scope for this.

**Comment by spencergilbert** [comment]
> Yes, those should be unified and response similarly. That is in scope for this.

Let's add it specifically then, since it's not actually related to the `/health` endpoint from an implementation aspect

**Comment by spencergilbert** [comment]
This probably also should be extended or created a related issue for spec'ing "component health"
